cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,376 @@
+          PORT:                         9443
+        Mounts:
+          /etc/config from config-volume (rw)
+          /var/run/secrets/eks.amazonaws.com/serviceaccount from aws-iam-token (ro)
+          /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-d22ld (ro)
+          /var/tls/cert from cert (ro)
+```
+
+Note: Since AWS IRSA works by mounting the service-account token to pods, the decision to use IRSA should be taken at install time. Credentials can be registered later on.
+
+**post installation**
+
+Once the installation is complete, the user can use rad env update to store relevant information for deploying resources to accounts and region associated with specific radius environment.
+
+```
+rad env update qa --aws-account-id <aws-account-id> --aws-region <aws-region> 
+```
+
+At this point, we can use aws AssumeRole to manage resources using environment specific account information. rp (for recipes) and ucp will have code changes that make sure of aws assumerole to be able to use the right credential specific to the radius environment for managing the aws resources.  
+
+```
+  client := sts.NewFromConfig(cfg)
+  assumeRoleArn := retrieveRoleARNCredential() //has logic to get the roleARN stored as credential
+  // Assume the new role
+  assumeRoleProvider := stscreds.NewAssumeRoleProvider(client, assumeRoleArn)
+  // Create a new config with the new credentials
+  cfg.Credentials = assumeRoleProvider
+  cfg.Region = region
+  // Create a new S3 service client
+  s3client := s3.NewFromConfig(cfg)
+  // Call the ListBuckets function
+  output, err := s3client.ListBuckets(context.TODO(), &s3.ListBucketsInput{})
+  :
+
+```
+
+### API design
+
+```
+@doc("AWS credential kind")
+enum AWSCredentialKind {
+  @doc("The AWS Access Key credential")
+  AccessKey,
+
+  @doc("The AWS IRSA credential")
+  IRSA,
+}
+```
+
+```
+model AwsIRSACredentialProperties extends AwsCredentialProperties {
+  @doc("Access Key kind")
+  kind: AWSCredentialKind.IRSA;
+
+  @doc("RoleARN for AWS IRSA identity")
+  @secret
+  roleARN: string;
+
+  @doc("The storage properties")
+  storage: CredentialStorageProperties;
+}
+```
+
+Based on this new model we have to update API calls. 
+
+### CLI Design
+
+* `rad init --full` should be updated to ask the user to choose between AWS access keys and IRSA. If IRSA is chosen, the user should be able to input a AWS IAM Role ARN. As a result of rad init, 
+   1. the helm charts should be updated with service-account token mounted as secret.
+   2. roleARN should be stored as a AWS credential
+
+* `rad install kubernetes --set global.aws.IRSA.enabled=true` should add the neccessary pod spec to mount the service-account token.
+
+* `rad credential register aws --access-key-id <access-key-id> --secret-access-key <secret-access-key>` should be replaced with 
+`rad credential register aws secret --access-key-id <access-key-id> --secret-access-key <secret-access-key>`
```

Comment:
This seems inconsistent from the interactive experience above where this option is named as "IAM access key". Any reason to name it as "secret" here